tests/python/examples/autodiff/test_simple_derivative.py

```diff
@@ -0,0 +1,44 @@
+import argparse
+
+
+def mkdir_p(dir_path):
+    '''Creates a directory. equivalent to using mkdir -p on the command line'''
+    from errno import EEXIST
+    from os import makedirs,path
+
+    try:
+        makedirs(dir_path)
+    except OSError as exc: # Python >2.5
+        if exc.errno == EEXIST and path.isdir(dir_path):
+            pass
+        else: raise
+
+
+def pic_simple_derivative(result_dir):
+    from taichi.examples.autodiff.simple_derivative import (initialize,
+                                                            xs, ys, grad_xs)
+    from matplotlib import pyplot as plt
```

Comment:
It's likely that we don't have matplotlib in CI test environment yet. So please feel free to add it in https://github.com/taichi-dev/taichi/blob/master/requirements_test.txt if you need it!

Reply:
The problem here is that the original example uses matplotlib, so importing from the example will result in error. I suggest adding it to `requirements_tests.txt`.

Reply:
get it, I will add matplotlib to `requirements_test.txt` in #4322 first